editoast/src/valkey_utils.rs

```diff
@@ -273,6 +273,65 @@ impl ValkeyConnection {
         });
         Ok(cached_values)
     }
+
+    /// Add one serializable member to a sorted set, or update its score if it already exists.
+    pub async fn json_zadd<
+        K: Debug + ToRedisArgs + Send + Sync,
+        S: ToRedisArgs + Send + Sync,
+        M: Serialize,
+    >(
+        &mut self,
+        key: K,
+        member: M,
+        score: S,
+    ) -> Result<()> {
+        let str_member = match serde_json::to_string(&member) {
+            Ok(member) => member,
+            Err(_) => {
+                return Err(RedisError::from((
+                    ErrorKind::IoError,
+                    "An error occurred serializing to json",
+                ))
+                .into());
+            }
+        };
+        self.zadd::<_, _, _, ()>(key, str_member, score).await?;
+        Ok(())
+    }
+
+    /// Return a range of members in a sorted set, by score.
+    pub async fn json_zrangebyscore<
+        T: DeserializeOwned,
+        K: ToRedisArgs + Send + Sync,
+        M: ToRedisArgs + Send + Sync,
+        MM: ToRedisArgs + Send + Sync,
+    >(
+        &mut self,
+        key: K,
+        min: M,
+        max: MM,
+    ) -> Result<Vec<T>> {
+        let serialized_members = self
+            .zrangebyscore::<_, _, _, Vec<String>>(key, min, max)
+            .await?;
+        if serialized_members.clone().is_empty() {
+            return Ok(vec![]);
+        }
```

Comment:
```suggestion
        if serialized_members.is_empty() {
            return Ok(serialized_members);
        }
```

that clone can potentially be very expensive

Reply:
Isn't the whole if kinda pointless? (Also, I don't think returning serialized member will work even if empty, since it's still a `Vec<String>`?)

Reply:
indeed

Reply:
Indeed, this part is useless because deserialized_members will be an empty vector if serialized_members is. Thanks for your reviews !